testsuite/tests/prometheus/apicast/selfmanaged/test_batching_caching_policy.py

```diff
@@ -72,33 +94,38 @@ def test_batcher_policy(prometheus, pod_monitor, api_client, staging_gateway, ap
     client = api_client()
 
     apicast_deployment = staging_gateway.deployment.name
+    labels = {"namespace": staging_gateway.openshift.project_name, "container": apicast_deployment}
+
+    prometheus.wait_on_next_scrape(apicast_deployment)
 
-    metrics_keys = prometheus.get_metrics(apicast_deployment)
+    metrics_keys = prometheus.get_metrics(labels=labels)
 
     assert "batching_policy_auths_cache_hits" not in metrics_keys
     assert "batching_policy_auths_cache_misses" not in metrics_keys
 
+    client.get("/get")
     for _ in range(NUM_OF_REQUESTS):
         client.get("/get")
 
     prometheus.wait_on_next_scrape(apicast_deployment, datetime.utcnow() + timedelta(seconds=BATCH_REPORT_SECONDS))
-    metrics_keys = prometheus.get_metrics(apicast_deployment)
+    metrics_keys = prometheus.get_metrics(labels=labels)
 
     assert "batching_policy_auths_cache_hits" in metrics_keys
     assert "batching_policy_auths_cache_misses" in metrics_keys
 
-    hits, misses = get_batching_metrics(prometheus, apicast_deployment)
+    hits, misses = get_batching_metrics(prometheus, apicast_deployment, staging_gateway.openshift.project_name)
 
-    assert int(hits) == NUM_OF_REQUESTS - 1
-    assert int(misses) == 1
+    assert (int(hits), int(misses)) == (NUM_OF_REQUESTS, 1)
```

Comment:
I think having both misses and hits to be compared at same time gives more info in case of failure, e.g.
```
    assert (int(hits), int(misses)) == (NUM_OF_REQUESTS - 1, 1)
E   assert (0, 5) == (4, 1)
E     At index 0 diff: 0 != 4
E     Full diff:
E     - (4, 1)
E     + (0, 5)
```